src/Employer.Web/Controllers/PreviewController.cs

```diff
@@ -24,10 +24,22 @@ public PreviewController(PreviewOrchestrator orchestrator)
             return View(vm);
         }
 
-        [HttpPost("vacancy-preview", Name = RouteNames.Preview_Index_Post)]
-        public IActionResult Index(IndexViewModel vm)
+        [HttpPost("vacancy-submit", Name = RouteNames.Preview_Submit_Post)]
+        public async Task<IActionResult> Submit(SubmitEditModel m)
         {
```

Comment:
If this action is Submit(Not preview), why do we need to post to it when the vacancyId is in the url. I think maybe a get would be suitable. That way on the preview page you can just choose to render a link for the button instead of the form. You already have a conditional around whether to show it.

Reply:
ok